diff -pruN ../libnatspec-0.2.6.orig/lib/data/charset_names.h ./lib/data/charset_names.h
--- ../libnatspec-0.2.6.orig/lib/data/charset_names.h	2010-01-11 05:19:58.000000000 +0600
+++ ./lib/data/charset_names.h	2011-04-11 09:43:11.000000000 +0700
@@ -34,6 +34,8 @@ static const struct charset_entry
     { "CP1256",    "cp1256",      1256 }, /* // */
     { "CP1257",    "cp1257",      1257 }, /* Estonian, Latvian, Lithuanian */
     { "CP1258",    "cp1258",      1258 }, /* // */
+	/* There is no IBM737 -> CP737 alias in iconv at the moment (glibc 2.12.1). */
+    { "CP737",     "cp737",        737 }, /* // 2.4/6 Codepage 737 (Greek)*/
     { "CP932",     "cp932",        932 }, /* Japanese charsets (Shift-JIS, EUC-JP) */
     { "CP936",     "cp936",        936 }, /* Simplified Chinese charset (CP936, GB2312) */
     { "CP949",     "cp949",        949 }, /* Korean charset (CP949, EUC-KR) */
@@ -45,10 +47,7 @@ static const struct charset_entry
     { "IBM424",    "cp424",        424 },
     { "IBM437",    "cp437",        437 }, /* // 2.4 cp Codepage 437 (United States, Canada) */
     { "IBM500",    "cp500",        500 },
-/* Unknown for iconv */
-/*    { "IBM737",    "cp737", 737 },*/ /* // 2.4/6 Codepage 737 (Greek)*/
-/*    { "IBM775",    "cp775", 775 },*/ /* // 2.4/6 Codepage 775 (Baltic Rim)*/
-
+    { "IBM775",    "cp775",        775 }, /* // 2.4/6 Codepage 775 (Baltic Rim)*/
     { "IBM850",    "cp850",        850 }, /* // 2.4 cp Codepage 850 (Europe)*/
     { "IBM852",    "cp852",        852 }, /* Codepage 852 (Central/Eastern Europe) */
     { "IBM855",    "cp855",        855 }, /* Codepage 855 (Cyrillic) */
diff -pruN ../libnatspec-0.2.6.orig/lib/data/gen_data.sh ./lib/data/gen_data.sh
--- ../libnatspec-0.2.6.orig/lib/data/gen_data.sh	2010-01-11 05:19:58.000000000 +0600
+++ ./lib/data/gen_data.sh	2011-04-11 09:43:11.000000000 +0700
@@ -3,7 +3,7 @@
 #    $Id: gen_data.sh,v 1.8 2005/03/09 20:07:30 lav Exp $
 
 OUTFILE=get_charset_data.h
-PRINTDATAPROGRAM="wwo ./print_data_string.exe.so"
+PRINTDATAPROGRAM="./print_data_string.exe"
 
 print()
 {
@@ -25,9 +25,9 @@ echo "Generating with WINE program..."
 echo "This is log error file. See for your problem locale here and send me a mail: lav@etersoft.ru">./gen_data.out.txt
 for i in `locale -a | LC_ALL=C sort`
 do
-	echo -e -n " {" >>$OUTFILE
+	echo -n " {" >>$OUTFILE
 	LANG=$i LC_CTYPE=$i $PRINTDATAPROGRAM 2>>./gen_data.out.txt >>$OUTFILE
-	echo -e " }," >>$OUTFILE
+	echo " }," >>$OUTFILE
 done
 echo >>$OUTFILE
 echo "/* Follow entries is dummy for ASCII/ANSI encoding */" >>$OUTFILE
diff -pruN ../libnatspec-0.2.6.orig/lib/data/get_charset_data.h ./lib/data/get_charset_data.h
--- ../libnatspec-0.2.6.orig/lib/data/get_charset_data.h	2010-01-11 05:19:58.000000000 +0600
+++ ./lib/data/get_charset_data.h	2011-04-11 09:45:55.000000000 +0700
@@ -105,10 +105,10 @@ static const struct charsetrel_entry cha
  {"de_LU.UTF8",   4103,  "UTF8",        "CP1252",   "IBM850",   "MAC" },
  {"de_LU@euro",   4103,  "ISO885915",   "CP1252",   "IBM850",   "MAC" },
  {"dz_BT",        1033,  "UTF8",        "CP1252",   "IBM437",   "MAC" },
- {"el_CY",        1032,  "ISO88597",    "CP1253",    NULL,       NULL },
- {"el_CY.UTF8",   1032,  "UTF8",        "CP1253",    NULL,       NULL },
- {"el_GR",        1032,  "ISO88597",    "CP1253",    NULL,       NULL },
- {"el_GR.UTF8",   1032,  "UTF8",        "CP1253",    NULL,       NULL },
+ {"el_CY",        1032,  "ISO88597",    "CP1253",   "CP737",     NULL },
+ {"el_CY.UTF8",   1032,  "UTF8",        "CP1253",   "CP737",     NULL },
+ {"el_GR",        1032,  "ISO88597",    "CP1253",   "CP737",     NULL },
+ {"el_GR.UTF8",   1032,  "UTF8",        "CP1253",   "CP737",     NULL },
  {"en_AG",        1033,  "UTF8",        "CP1252",   "IBM437",   "MAC" },
  {"en_AU",        3081,  "ISO88591",    "CP1252",   "IBM850",   "MAC" },
  {"en_AU.UTF8",   3081,  "UTF8",        "CP1252",   "IBM850",   "MAC" },
@@ -182,9 +182,9 @@ static const struct charsetrel_entry cha
  {"es_UY.UTF8",   14346, "UTF8",        "CP1252",   "IBM850",   "MAC" },
  {"es_VE",        8202,  "ISO88591",    "CP1252",   "IBM850",   "MAC" },
  {"es_VE.UTF8",   8202,  "UTF8",        "CP1252",   "IBM850",   "MAC" },
- {"et_EE",        1061,  "ISO88591",    "CP1257",    NULL,       NULL },
- {"et_EE.ISO885915", 1061,  "ISO885915",   "CP1257",    NULL,       NULL },
- {"et_EE.UTF8",   1061,  "UTF8",        "CP1257",    NULL,       NULL },
+ {"et_EE",        1061,  "ISO88591",    "CP1257",   "IBM775",       NULL },
+ {"et_EE.ISO885915", 1061,  "ISO885915",   "CP1257",   "IBM775",    NULL },
+ {"et_EE.UTF8",   1061,  "UTF8",        "CP1257",   "IBM775",    NULL },
  {"eu_ES",        1069,  "ISO88591",    "CP1252",   "IBM850",   "MAC" },
  {"eu_ES.UTF8",   1069,  "UTF8",        "CP1252",   "IBM850",   "MAC" },
  {"eu_ES@euro",   1069,  "ISO885915",   "CP1252",   "IBM850",   "MAC" },
@@ -278,10 +278,10 @@ static const struct charsetrel_entry cha
  {"li_BE",        1033,  "UTF8",        "CP1252",   "IBM437",   "MAC" },
  {"li_NL",        1033,  "UTF8",        "CP1252",   "IBM437",   "MAC" },
  {"lo_LA",        1033,  "UTF8",        "CP1252",   "IBM437",   "MAC" },
- {"lt_LT",        1063,  "ISO885913",   "CP1257",    NULL,       NULL },
- {"lt_LT.UTF8",   1063,  "UTF8",        "CP1257",    NULL,       NULL },
- {"lv_LV",        1062,  "ISO885913",   "CP1257",    NULL,       NULL },
- {"lv_LV.UTF8",   1062,  "UTF8",        "CP1257",    NULL,       NULL },
+ {"lt_LT",        1063,  "ISO885913",   "CP1257",   "IBM775",    NULL },
+ {"lt_LT.UTF8",   1063,  "UTF8",        "CP1257",   "IBM775",    NULL },
+ {"lv_LV",        1062,  "ISO885913",   "CP1257",   "IBM775",    NULL },
+ {"lv_LV.UTF8",   1062,  "UTF8",        "CP1257",   "IBM775",    NULL },
  {"mai_IN",       1033,  "UTF8",        "CP1252",   "IBM437",   "MAC" },
  {"mg_MG",        1033,  "ISO885915",   "CP1252",   "IBM437",   "MAC" },
  {"mg_MG.UTF8",   1033,  "UTF8",        "CP1252",   "IBM437",   "MAC" },
